### DISCUSSION TITLE
Switch locale with nested params route

### DISCUSSION THREAD

**Discussion by Mimachh** [amannn/next-intl#969] [Q&A]
### Description

Hello, your job is really amazing.
However I'm facing an issue. I actually have nested route params "/store/[storeId]/products/[id]", and the locale switcher does not work as excepted. I tried this
       router.replace( {pathname, params}, {locale: nextLocale} );

But it does not redirect me well. Have you got solution ?

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

App router

### Reproduction description

Steps to reproduce:

I only install a fresh nextjs 14 app router, with Next Intl + localized pathnames.

When I click on the github example of locale switcher I'm not redirected to the right pathname. I'm redirected to the url without the last params. 
Eg: /dashboard/store/[storeId]/products/[id] -> /tableau-de-bord/boutique/[storedId]      <-- I lost products and params.

### Expected behaviour

I would have the correct localized pathname. Thank for your job.

**Comment by amannn** [comment]
I'm moving this to a discussion since it seems to be a usage question. Note that bug reports require a reproduction that illustrates that you're experiencing a bug that is caused by `next-intl` and not something that is misconfigured in your app.

**Reply by Mimachh** [reply]
Well, it's not a misconfiguration from my app, at least I don't think so. The local switcher redirect me to the good url when I use only one params in the pathname but not with 2 or more.
Is the router.replace( {pathname, params}, {locale: nextLocale} ); should works with more than 1 params?
